geo/src/algorithm/stitch.rs

```diff
@@ -0,0 +1,590 @@
+use std::collections::HashMap;
+
+use geo_types::{Coord, Line, LineString, MultiPolygon, Polygon, Triangle};
+
+use crate::winding_order::WindingOrder;
+use crate::{Contains, GeoFloat};
+
+// ========= Error Type ============
+
+#[derive(Debug)]
+pub enum LineStitchingError {
+    IncompleteRing(&'static str),
+    InvalidGeometry(&'static str),
+    MissingParent,
+    NoExtremum,
+}
+
+impl std::fmt::Display for LineStitchingError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+impl std::error::Error for LineStitchingError {}
+
+pub(crate) type TriangleStitchingResult<T> = Result<T, LineStitchingError>;
+
+// ========= Main Algo ============
+
+/// Trait to stitch together split up triangles.
+pub trait StitchTriangles<T: GeoFloat> {
+    /// This stitching only happens along identical edges which are located in two separate
+    /// geometries. Please read about the required pre conditions of the inputs!
+    ///
+    /// ```text
+    /// ┌─────x        ┌─────┐
+    /// │    /│        │     │
+    /// │   / │        │     │
+    /// │  /  │  ───►  │     │
+    /// │ /   │        │     │
+    /// │/    │        │     │
+    /// x─────┘        └─────┘
+    /// ```
+    ///
+    /// # Pre Conditions
+    ///
+    /// - All the given triangles must be unique. If you're not sure about that, deduplicate before
+    ///   calling the algorithm!
+    /// - Input triangles should be valid polygons. For a definition of validity
+    ///   c.f. https://www.postgis.net/workshops/postgis-intro/validity.html
```

Comment:
Fixing a doc lint suggestion

```suggestion
    ///   c.f. <https://www.postgis.net/workshops/postgis-intro/validity.html>
```

Reply:
Thanks for doing that in the mean time already 🙏🏼